beta/src/content/docs/en/guides/server-side-rendering.md

```diff
@@ -0,0 +1,289 @@
+  // 1. create separate instance of `effector`'s state - special `Scope` object
+  const scope = fork({
+    values: [
+      // some parts of app's state can be immideatly set to relevant states,
+      // before any computations started
+      [$pathname, req.pathname],
+    ],
+  });
+
+  // 2. start app's logic - all computations will be performed according to the model's logic,
+  // as well as any required effects
+  await allSettled(appStarted, {
+    scope,
+  });
+
+  // 3. Serialize the calculated state, so it can be passed over the network
+  const storesValues = serialize(scope);
+
+  // 4. Render the app - also into some serializable version
+  const app = renderToString(
+    // by using Provider with the scope we tell the <App />, which state of the stores it should use
+    <Provider value={scope}>
+      <App />
+    </Provider>,
+  );
+
+  // 5. prepare serialized html response
+  //
+  // This is serialization (or network) boundary
+  // The point, where all state is stringified to be sent over the network
+  //
+  // `effector`s state is stored as a `<script>`, which will set the state into global object
+  // `react`'s state is stored as a part of the DOM tree.
+  return `
+    <html>
+      <head>
+        <script>
+          self._SERVER_STATE_ = ${JSON.stringify(storesValues)}
+        </script>
+        <link rel="stylesheet" href="styles.css" />
+        <script defer src="app.js" />
+      </head>
+      <body>
+        <div id="app">
+          ${app}
+        </div>
+      </body>
+    </html>
+  `;
+}
+```
+
+☝️ In this code we have created the `html` string, which user will receive over the network and which contains serialized state of the whole app.
+
+## Client entrypoint
+
+When the generated html string reaches the client browser, has been processed by the parser and all the required assemblies have been loaded - our application code starts working on the client.
+
+At this point `<App />` needs to restore its past state (which was computed on the server), so that it doesn't start from scratch, but starts from the same point the work reached on the server.
+
+The process of restoring the server state at the client is usually called **hydration** and this is what client entrypoint should actually do:
+
+```tsx
+// client.tsx
+import React from "react";
+import { hydrateRoot } from "react-dom/client";
+import { fork, allSettled } from "effector";
+import { Provider } from "effector-react";
+import { App, appStarted } from "effector";
+
+/**
+ * 1. Find, where the server state is stored and retreive it
+ *
+ * See the server handler code to find out, where it was saved in the html
+ */
+const effectorState = globalThis._SERVER_STATE_;
+const reactRoot = document.querySelector("#app");
+
+/**
+ * 2. Initiate the client scope of `effector` with server-calculated values
+ */
+const clientScope = fork({
+  values: effectorState,
+});
+
+allSettled(appStarted, { scope: clientScope });
```

Comment:
🚧 `appStarted` have to be called **after** react hydration. Otherwise, it could lead to races and html-state mismatches.